### DISCUSSION TITLE
Mapping between azurerm res and AVM module

### DISCUSSION THREAD

**Discussion by alefteris** [Azure/Azure-Verified-Modules#1451] [Q&A]
Hi all. Is there an existing map or a way to create a mapping between an azurerm resource name and the corresponding AVM module name, if one exists for the azurerm resource?

In the AVM git repo, there is a [CSV that lists all the Terraform AVM resource modules](https://github.com/Azure/Azure-Verified-Modules/blob/main/docs/static/module-indexes/TerraformResourceModules.csv) and has a ProviderNamespace and ResourceType column. But I haven't found a way to match this against an azurerm resource. Is there any place in the azurerm provider code that defines what the namespace and provider is for a resource?

This mapping could be used to figure out how many azurerm resources can be replaced in a terraform project by the currently published AVM modules or what needed AVM modules are missing. Also it could be used in a linter to suggest the use of the AVM module instead of the azurerm resource.

Thanks.

**Comment by matt-FFFFFF** [comment]
The only way I can think of it to scour the generated README files for each module. They contain a list of deployed resources.

**Reply by alefteris** [reply]
The problem is that the module READMEs list a lot more Terraform resources than the "primary" one. It would not be possible to filter out just the primary Terraform resource out of all those I think.

**Comment by alefteris** [comment]
I found this repo https://github.com/magodo/aztft (by an MS employee and also used as a dependency from https://github.com/Azure/aztfexport) that could be used to do the mapping.
